OVERVIEW.md

```diff
@@ -0,0 +1,123 @@
+3. It provides best practices and guidance on how to organize code and utilize Hydra for research, production, and other innovative use cases.
+
+##### Terminology for this overview:
+- The overall effort and repo is named:`hydra-torch`
+- **Library:** The library being configured
+  - e.g. `torch`, `torchvision`
+- **Project:** The project corresponding to the library being configured
+  - e.g. `hydra-configs-torch`, `hydra-configs-torchvision`
+- **Package:** The installable package containing the configs
+  - here, also `hydra-configs-torch`, `hydra-configs-torchvision`. find definition in the project dir's `setup.py`.
+
+
+
+### Repo Structure
+```
+📂 hydra-torch
+└ 📁 configen 				# source files used when generating package content. all projects have their own subdirectory here.
+└ 📁 hydra-configs-torch		# a project corresponding to configuring a specific library (contains package definition)
+└ 📁 hydra-configs-torchvision		# "
+└ 📁 hydra-configs-<future-library> 	# "
+└ 📁 examples				# use-cases and tutorials
+```
+
+Each `hydra-configs-<library-name>` defines its own package corresponding to a project it provides classes for. That is, `hydra-configs-torch` contains a package (with its own `setup.py`) which provides classes for `torch`.
+
+[`hydra-configs-projects.txt`](hydra-configs-projects.txt) specifies a list of projects this repo maintains. When adding a project, please update this file.
+
+
+### Packaging
+
+This repo maintains multiple packages. An important area of consideration for these packages is to organize them such that they are compatible with each other and have an 'expected', intuitive structure. The easiest way to do this is to enforce a standard for all Hydra configs packages to follow.
+
+Our approach makes use of [Native Namespace Packages](https://packaging.python.org/guides/packaging-namespace-packages/#native-namespace-packages). This allows all packages to share the top level namespace `hydra_configs`. The only real requirement for this package format is to ensure there is no `__init__.py` located in the namespace folder.
+
+The format for folder structure per project is always:
+```
+📂 hydra-configs-<library-name>     # the dir containing the package of configs for <library>
+├ 📁 hydra_configs		    # the namespace we will always use
+│ └ 📁 <library-name>		    # e.g. 'torchvision'
+│   └ 📁 <library-subpackage-name>  # e.g. 'transforms'
+│      ⋮
+│      └ <module>.py                # e.g. 'transforms.py'
+└ setup.py 			    # configures this package for setuptools
+```
+
+The beauty of this approach is that users can be sure the importing idiom is reliably:
+```python
+from hydra_configs.<project_name>.path.to.module import <ClassName>Conf
+```
+
+This will retain compatibility even with repositories that define configs for hydra outside of the `hydra-torch` repo as long as they follow this pattern.
```

Comment:
what compatibility?

Reply:
Need a good way to point out that using the namespace package design lets others create packages that are accessible using the same top-level namespace. This is what I meant by compatibility. We'll work on the wording to indicate this better.

Reply:
I think we should probably have a small section about creating similar projects (inside or outside of this repo).
It's secondary at this time.